### PR TITLE
Add prerender docs index

### PR DESCRIPTION
**Pull request by diana-macrometa**
This PR adds an index page for Prerendering that explains what the service is, how it works, and lists the features. This content is taken directly from Notion with some minor edits.

### CODE REVIEW

File: docs/_photoniq/prerendering/index.md [hunk 1]
```diff
@@ -3,4 +3,32 @@ sidebar_position: 1
...
 
+### How It Works
+
+Prerendering works by running a full version of a web browser, such as Chrome, in a server environment. This browser loads the entire web application, including executing all JavaScript, just like a user's browser would. Once the JavaScript has finished executing, the prerendering service takes a snapshot of the resulting HTML and serves this static version to the client.
+
+This process is transparent to the end user. When a user or a search engine crawler makes a request to a website, the server determines whether to serve the dynamic or prerendered version of the site. This decision can be based on various factors, such as the type of user agent making the request.
+
+## Features
+
+Key capabilities include:
```
Comment (diana-macrometa) [inline docs/_photoniq/prerendering/index.md]:
```suggestion
Key capabilities of Dynamic Prerendering include:
```

File: docs/_photoniq/prerendering/index.md [hunk 2]
```diff
@@ -3,4 +3,32 @@ sidebar_position: 1
 title: Dynamic Prerendering
 ---
 
+The PhotonIQ Dynamic Prerendering service is an innovative solution that generates static HTML pages from dynamic websites or web applications. This process, known as prerendering, is particularly useful for improving the search engine optimization (SEO) of websites that rely heavily on JavaScript.
 
+### How It Works
+
+Prerendering works by running a full version of a web browser, such as Chrome, in a server environment. This browser loads the entire web application, including executing all JavaScript, just like a user's browser would. Once the JavaScript has finished executing, the prerendering service takes a snapshot of the resulting HTML and serves this static version to the client.
+
+This process is transparent to the end user. When a user or a search engine crawler makes a request to a website, the server determines whether to serve the dynamic or prerendered version of the site. This decision can be based on various factors, such as the type of user agent making the request.
```
Comment (Luba89) [inline docs/_photoniq/prerendering/index.md]: "When a user or a search engine crawler makes a request to a website, the server determines whether to serve the dynamic or prerendered version of the site. This decision can be based on various factors, such as the type of user agent making the request."
This was something that was planned in the beginning, but now, we are just prerendering. You need to send proper headers, and we will prepare the URL for you. The customer's job is to indefinite the bot and route it on our service.

File: docs/_photoniq/prerendering/index.md [hunk 3]
```diff
@@ -3,4 +3,32 @@ sidebar_position: 1
...
+
+- **Caching**: Boost speed by storing results from resource-intensive operations or retaining static resources for immediate retrieval.
+- **Load Balancing**: Evenly distribute incoming traffic across multiple servers. This balance ensures no single server gets overburdened, promoting consistent application reliability.
+- **Scalability**: Tailored for large-scale prerendering challenges. The service uses a distributed compute strategy, potentially running numerous instances in a given region to manage high-traffic demands.
+
+### Reliability and Security
+
+- **Fault Tolerance and Resiliency**: Built to address server or network disruptions. In the event of network issues, the service opts to serve older, reliable data over rejecting the request.
+- **Rate Limiting**: Implement limits on client request frequency to deter misuse and ensure everyone gets a fair share of the resources.
+- **Authorization**: Access and configuration of the prerender service are safeguarded, requiring user authorization through JWT or API keys.
```
Comment (Luba89) [inline docs/_photoniq/prerendering/index.md]: JWT and API keys are currently not supported. The customer is using a special header to use the service

Comment (diana-macrometa) [reply docs/_photoniq/prerendering/index.md]: How can I rephrase that? "Access and configuration of the prerender service are safeguarded, requiring special headers for security."?

File: docs/_photoniq/prerendering/index.md (hunk 2 above)
Comment (diana-macrometa) [reply docs/_photoniq/prerendering/index.md]: Should I just remove the whole paragraph, or do we want to reword it?

Comment (ghost) [reply docs/_photoniq/prerendering/index.md]: @diana-macrometa could you please reword it according to @Luba89 text? Thanks!

File: docs/_photoniq/prerendering/index.md (hunk 3 above)
Comment (ghost) [reply docs/_photoniq/prerendering/index.md]: @diana-macrometa we can just ditch `requiring user authorization through JWT or API keys.`, we will have a login page in the future

**Review by ghost — CHANGES_REQUESTED** [review]
Minor things to change

File: docs/_photoniq/prerendering/index.md (hunk 2 above)
Comment (diana-macrometa) [reply docs/_photoniq/prerendering/index.md]: How do you "indefinite a bot"?

Comment (diana-macrometa) [reply docs/_photoniq/prerendering/index.md]: Guys, I do not know how to translate "You need to send proper headers, and we will prepare the URL for you. The customer's job is to indefinite the bot and route it on our service." into something appropriate for the user.

"When a user or a search engine crawler makes a request to a website,..." what happens? I know that someone has headers somewhere and there are prerendered pages that are getting served somehow.

File: docs/_photoniq/prerendering/index.md (hunk 3 above)
Comment (diana-macrometa) [reply docs/_photoniq/prerendering/index.md]: Updated. WDYT?

File: docs/_photoniq/prerendering/index.md (hunk 2 above)
Comment (Luba89) [reply docs/_photoniq/prerendering/index.md]: @diana-macrometa @dlozina-macrometa, Sorry for the confusion,  I think we can remove it, as we are not using that approach anymore. Currently, we are just receiving requests, and we are then prerendering; we are not checking if someone is a bot or not; that logic now is on the client side.

Comment (diana-macrometa) [reply docs/_photoniq/prerendering/index.md]: Done. Btw, I took that text straight out of Notion, so you guys might want to update that.

**Comment by diana-macrometa** [comment]
Reviewers - Are we ready for approval?

**Review by Luba89 — APPROVED** [review]

**Comment by diana-macrometa** [comment]
@dlozina-macrometa, I need your approval to unblock this PR and merge.

**Review by ghost — APPROVED** [review]
👍

**Pull request merged**